File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let len = nums.length;
    for(let i = 0; i < len; i++) {
        let j = i + 1;
        while (j < len)  {
            if(nums[i] + nums[j] == target) {
                return [i, j];
            }
            j += 1;
        }
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for(int j = 0; j < nums.length; j++) {
            if(map.containsKey(target - nums[j])) {
                return new int[]{j, map.get(target - nums[j])};
            } else {
                map.put(nums[j], j);
            }
        }
        return new int[2];
    }
}